shared/p2p/negotiation.go

```diff
@@ -0,0 +1,50 @@
+package p2p
+
+import (
+	"context"
+
+	ggio "github.com/gogo/protobuf/io"
+	host "github.com/libp2p/go-libp2p-host"
+	inet "github.com/libp2p/go-libp2p-net"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+const handshakeProtocol = prysmProtocolPrefix + "/handshake"
+
+func setupPeerNegotiation(h host.Host, contractAddress string) {
+	h.Network().Notify(&inet.NotifyBundle{
+		ConnectedF: func(net inet.Network, conn inet.Conn) {
+			log.Debug("Checking connection to peer")
+
+			s, err := h.NewStream(context.Background(), conn.RemotePeer(), handshakeProtocol)
```

Comment:
@raulk, this is blocking / never opens the stream. Am I missing something?

Reply:
Could you try to create the stream in a goroutine? "never block the callback" is the problem I think you're seeing here.

Reply:
That works, thanks!